scripts/import/import_EPMC.pl

```diff
@@ -528,52 +574,78 @@ sub report_summary{
                                          and p1.pmid is not null
                                        ]);
 
-    my $dup2_ext_sth = $dba->dbc->prepare(qq[ select p1.publication_id, p2.publication_id, p2.pmcid
-                                         from publication p1, publication p2
-                                         where p1.pmcid = p2.pmcid
-                                         and p1.publication_id < p2.publication_id
-                                       ]);
-
-   my $dup3_ext_sth = $dba->dbc->prepare(qq[ select p1.publication_id, p2.publication_id, p2.doi
-                                         from publication p1, publication p2
-                                         where p1.doi = p2.doi
-                                         and p1.publication_id < p2.publication_id
-                                       ]);
-
-
-
-
+    # Query is taking a few hours to run
+    # my $dup2_ext_sth = $dba->dbc->prepare(qq[ select p1.publication_id, p2.publication_id, p2.pmcid
+    #                                      from publication p1, publication p2
+    #                                      where p1.pmcid = p2.pmcid
+    #                                      and p1.publication_id < p2.publication_id
+    #                                      and p1.pmcid is not null
+    #                                    ]);
+
+   my $dup2_ext_sth = $dba->dbc->prepare(qq[ SELECT publication_id, pmcid 
+                                             FROM publication 
+                                             GROUP BY pmcid HAVING COUNT(*) > 1 and pmcid is not null ]);
+
+   # Query is taking a few hours to run 
+   # my $dup3_ext_sth = $dba->dbc->prepare(qq[ select p1.publication_id, p2.publication_id, p2.doi
+   #                                       from publication p1, publication p2
+   #                                       where p1.doi = p2.doi
+   #                                       and p1.publication_id < p2.publication_id
+   #                                       and p1.doi is not null
+   #                                     ]);
+
+    my $dup3_ext_sth = $dba->dbc->prepare(qq[ SELECT publication_id, doi 
```

Comment:
Suggest putting publication_id in GROUP BY and the doi not null in WHERE clause

Reply:
Similar comment for doi duplication checking as for pmcid - this group by will not pick up duplicate doi.
Initial comment putting publication_id in group by was because of columns named in select not in group by.  If done with initial group by, the publication_id is non deterministic